### PR TITLE
TW-369: added kb for namespace setup considerations

### PR DESCRIPTION
**Pull request by ktenzer**
## What does this PR do?
Document namespace setup considerations in KB article

### CODE REVIEW

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+- Namespaces are usually defined per use case. A use case can encompass a broad range of workflow types, and a nearly unlimited scale of concurrent workflow executions.
+- Namespaces can be split along additional boundaries such as service, application, domain or even sub-domain.
+- Environments such as production and development usually have requirements for isolation. It is recommended that each environment has its own namespace.
+- Namespaces should be used to reduce blast radius for mission critical applications.
+- Workflows that need to communicate with one another (for now) should be in the same namespace.
+- Careful consideration should be taken when sharing namespaces across team or domain boundaries.
+
+## Examples
+
+Below are some ideas about how to organize namespaces.
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:43]:
```suggestion
Following are some ideas about how to organize Namespaces.
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+### Example 1: Namespace per Use Case/Environment
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:45]:
```suggestion
### Example 1: Namespace per use case and environment
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+A namespace per use case/environment is recommended for simple configurations where multiple services, team or domain boundaries don't exist.
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:47]:
```suggestion
We recommend using one Namespace for each use case and environment combination for simple configurations in which multiple services and team or domain boundaries don't exist.
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+Sample naming convention:
+
+<pre>
+&lt;user case>_&lt;environment>
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:52]:
```suggestion
&lt;use-case>_&lt;environment>
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+</pre>
+
+### Example 2: Namespace per Use Case/Service/Environment
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:55]:
```suggestion
### Example 2: Namespace per use case, service, and environment
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+A namespace per use case/service/environment is recommended when multiple services that are part of same use case, communicate externally to Temporal via API (HTTP/gRPC).
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:57]:
```suggestion
We recommend using one Namespace for each use case, service, and environment combination when multiple services that are part of same use case communicate externally to Temporal via API (HTTP/gRPC).
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+Sample naming convention:
+
+<pre>
+&lt;user case>_&lt;service>_&lt;environment>
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:62]:
```suggestion
&lt;use-case>_&lt;service>_&lt;environment>
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+</pre>
+
+### Example 3: Namespace per Use Case/Domain/Environment
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:65]:
```suggestion
### Example 3: Namespace per use case, domain, and environment
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+A namespace per use case/domain/environment is recommended when multiple services that are part of the same use case need to communicate with one another, via signals or starting of child workflows. It is very important however, in this case, to be mindful about workflowId uniqueness by prefixing workflowId with a service specific string. If multiple teams are involved the domain could also represent a team boundary.
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:67]:
```suggestion
We recommend using one namespace per use case, domain, and environment combination when multiple services that are part of the same use case need to communicate with each another via [Signals](/concepts/what-is-a-signal) or by starting [Child Workflows](/concepts/what-is-a-child-workflow).
In this case, though, you must be mindful about Workflow Id uniqueness by prefixing each Workflow Id with a service-specific string.
If multiple teams are involved, the domain could also represent a team boundary.
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+
+Sample naming convention:
+
+<pre>
+&lt;user case>_&lt;domain>_&lt;environment>
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:72]:
```suggestion
&lt;use-case>_&lt;domain>_&lt;environment>
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+</pre>
+
+Sample workflowId convention:
+
+<pre>
+&lt;service string>_&lt;workflowId>
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:78]:
```suggestion
&lt;service-string>_&lt;workflow-id>
```

File: kb/temporal-cloud-namespace-setup-considerations.md (hunk 9 above)
Comment (tsurdilo) [inline kb/temporal-cloud-namespace-setup-considerations.md:67]: > to be mindful about workflowId uniqueness

and task queue names.

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+Sample naming convention:
+
+<pre>
+&lt;use-case>_&lt;service>_&lt;environment>
+</pre>
+
+### Example 3: Namespace per use case, domain, and environment
+
+We recommend using one namespace per use case, domain, and environment combination when multiple services that are part of the same use case need to communicate with each another via [Signals](/concepts/what-is-a-signal) or by starting [Child Workflows](/concepts/what-is-a-child-workflow).
+In this case, though, you must be mindful about Workflow Id uniqueness by prefixing each Workflow Id with a service-specific string.
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:79]:
```suggestion
In this case, though, you must be mindful about Workflow Id uniqueness by prefixing each Workflow Id with a service-specific string.
The name of each Task Queue must also be unique.
```

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+Sample naming convention:
+
+<pre>
+&lt;use-case>_&lt;service>_&lt;environment>
+</pre>
+
+### Example 3: Namespace per use case, domain, and environment
+
+We recommend using one namespace per use case, domain, and environment combination when multiple services that are part of the same use case need to communicate with each another via [Signals](/concepts/what-is-a-signal) or by starting [Child Workflows](/concepts/what-is-a-child-workflow).
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:78]:
```suggestion
We recommend using one namespace per use case, domain, and environment combination when multiple services that are part of the same use case need to communicate with each another via [Signals](/workflows#signal) or by starting [Child Workflows](/workflows#child-workflow).
```

**Review by djmagee — COMMENTED** [review]
Changing links that aren't converted by the Assembly Workflow.

File: kb/temporal-cloud-namespace-setup-considerations.md [hunk 14]
```diff
@@ -0,0 +1,93 @@
+---
+slug: cloud-namespace-naming-considerations
+title: Temporal Cloud Namespace naming considerations
+tags:
+  - cloud
+  - namespaces
+date: 2023-03-10T00:00:00Z
```
Comment (djmagee) [inline kb/temporal-cloud-namespace-setup-considerations.md:7]:
```suggestion
date: 2023-04-20T00:00:00Z
```

**Review by djmagee — APPROVED** [review]

**Pull request merged**